### PR TITLE
Add llamaindex agent integration

### PR DESCRIPTION
**Pull request by colombod**
Adding contrib support for integrating llamaidnex agentrunner in autogen architecture

### CODE REVIEW

File: website/blog/authors.yml
Comment (sonichi) [inline website/blog/authors.yml:1]: No need to change this file.

File: test/test_notebook.py
Comment (sonichi) [inline test/test_notebook.py:1]: no need to change this file.

**Review by sonichi — COMMENTED** [review]
Could you please write a test and consider adding a notebook example to demonstrate it? The test can be added to contrib-test.yml.

File: test/test_notebook.py
Comment (colombod) [reply test/test_notebook.py:1]: changes where done by running the pre-commit tool

**Comment by colombod** [comment]
pushed notebook and test

**Comment by sonichi** [comment]
Also, consider updating https://microsoft.github.io/autogen/docs/ecosystem

File: notebook/agentchat_group_chat_with_llamaindex_agents.ipynb [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+ ],
+ "metadata": {
+  "front_matter": {
+   "description": "Equip your agent with functions that can efficiently implement features into your software application.",
+   "tags": [
+    "function call",
+    "code generation",
+    "tool use",
+    "software engineering"
+   ]
```
[the remark this answers is not in the document]
Comment (colombod) [reply notebook/agentchat_group_chat_with_llamaindex_agents.ipynb]: done, how do you folks manage that? Editing the json directly?

File: .github/workflows/contrib-openai.yml [hunk 2]
```diff
@@ -416,3 +416,37 @@ jobs:
...
           file: ./coverage.xml
           flags: unittests
+
+  LlamaIndexAgent:
+    runs-on: ${{ matrix.os }}
+    environment: openai1
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-2019]
```
[the remark this answers is not in the document]
Comment (colombod) [reply .github/workflows/contrib-openai.yml]: done!

File: notebook/agentchat_group_chat_with_llamaindex_agents.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (sonichi) [reply notebook/agentchat_group_chat_with_llamaindex_agents.ipynb]: I do that. Maybe there's a better way.

[the remark this answers is not in the document]
Comment (colombod) [reply notebook/agentchat_group_chat_with_llamaindex_agents.ipynb]: done!

**Comment by colombod** [comment]
Closing this as this work is now tracked on #2831

**Pull request closed without merging**